### ISSUE TITLE
Crash on macOS Monterey

### ISSUE THREAD

**Issue by Casuallynoted** [Sonarr/Sonarr#4718]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Sonarr appears to crash within five minutes of running on macOS 12 Monterey. I've tried updating to the new Mono RC but this does not seem to fix it. Notably, this issue does not occur with Radarr or Lidarr.

### Expected Behavior

Sonarr should remain open as it does in macOS 11 Big Sur

### Steps To Reproduce

1. Update to macOS Monterey on any Mac
2. Run Sonarr, observe that it closes shortly after opening with seemingly no info in debug/trace logs

### Environment

```markdown
- OS: macOS 12 Monterey
- Sonarr: 3.0.6.1342
- Docker Install: No
- Using Reverse Proxy: Yes
- Browser: Safari/Chrome/Firefox
```

### What branch are you running?

Main

### Anything else?

[sonarr.trace.17.txt](https://github.com/Sonarr/Sonarr/files/7422248/sonarr.trace.17.txt)
I've included the trace log but see nothing identifying in it that could point to what's causing this to happen.

**Comment by nojkepop** [comment]
> > Where exactly are we supposed to be placing the libsqlite3.dylib file on MacOS?
> 
> Go to applications, then right click Sonarr and select 'Show Package Contents'. Navigate through to Contents>MacOS>bin Delete the old libsqlitex.dylib and drop in the new linked one - you will also see its alongside the Sonarr.exe file as per instructions :) Startup Sonarr again, all seems to be working for me.

I followed these instructions and the issue appears to be resolved. The only thing I did differently was I trashed two files libsqlite3.0dylib and libsqlite3.dylib and then replaced it with the linked file found here - https://download.sonarr.tv/hotfixes/libsqlite3.0.dylib

Sonarr was crashing within a few minutes of opening before and its now been running for a few days without issue.

**Comment by jchhenderson** [comment]
> > Go to applications, then right click Sonarr and select 'Show Package Contents'. Navigate through to Contents>MacOS>bin Delete the old libsqlitex.dylib and drop in the new linked one - you will also see its alongside the Sonarr.exe file as per instructions :) Startup Sonarr again, all seems to be working for me.
> 
> I followed these instructions and the issue appears to be resolved. The only thing I did differently was I trashed two files libsqlite3.0dylib and libsqlite3.dylib and then replaced it with the linked file found here - https://download.sonarr.tv/hotfixes/libsqlite3.0.dylib
> 
> Sonarr was crashing within a few minutes of opening before and its now been running for a few days without issue.

Thanks!  ...this appears to have fixed it for me as well 👍

**Comment by Taloth** [comment]
Can one of you switch to the `mac-m1` release channel in Settings->General? I updated libmediainfo and bundled a (newer) libsqlite library as well. That should deal with both x64 and arm64 (m1) chips.

Please report any issues here. But when you reply with an issue, please post `mono --version` if you haven't done so earlier, so I know which arch you're running on.

**Comment by whodean** [comment]
To be clear, will this channel work with an Intel Mac right now as well as a M1?

**Comment by Taloth** [comment]
Yup, both. Only thing that won't work is old 32 bit systems, and I don't think those exist anymore :smile:

**Comment by jchhenderson** [comment]
done and updated to 3.0.6.1374 (Mac-m1).  Will let you know how it goes.

...I am on Intel.

**Comment by Taloth** [comment]
cool, thx! Lemme know in a day or two (or longer) whether imports work properly. That covers both mediainfo and sqlite.

**Comment by whodean** [comment]
So far so good!

**Comment by ekko20six** [comment]
have updated to the Mac-m1 version now 3.0.6.1374 and mono is reported as 6.12.0.90 within Sonarr and terminal (was already at this mono version before the Sonarr update to Mac-m1 though)
will report back findings

**Comment by jchhenderson** [comment]
Import worked fine for me just now...

**Comment by ekko20six** [comment]
oh dear, rebooted computer last night and now Sonarr doesn't want to play anymore.
Woke up to:
Connection Lost
Sonarr has lost its connection to the backend and will need to be reloaded to restore functionality
Sonarr will try to connect automatically or you can click reload below.


Reload doesn't work, need to restart from applications folder, and while it is updating/refreshing it crashed again.
What other info would you like to troubleshoot.
Am using a 
Mac mini (2018) with Monterey 12.0.1

**Comment by Taloth** [comment]
> What other info would you like to troubleshoot.

Console output. So we know where and how it's crashing.

**Comment by ekko20six** [comment]
looks like its mono that is crashing. The attached is from me restarting Sonarr to confirm the behaviour.
There was a crash at 7:21am and 12:01am as well
[Mono-crash.txt](https://github.com/Sonarr/Sonarr/files/7493053/Mono-crash.txt)
ng

**Comment by Taloth** [comment]
Hm, there's a surprising lack of crash origin there. Can you grab sonarr's trace level logs too? See what it was doing just before the crash.

**Comment by whodean** [comment]
I had a crash overnight too

Does this give you enough info?

`-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               mono-sgen64 [15636]
Path:                  /Applications/Sonarr.app/Contents/MacOS/Sonarr
Identifier:            mono-sgen64
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2021-11-06 10:42:53.1674 -0400
OS Version:            macOS 12.0.1 (21A559)
Report Version:        12
Bridge OS Version:     6.0 (19P548)
Anonymous UUID:        712D7E3A-BC86-653C-8631-4F57F7D7FED4


Time Awake Since Boot: 690000 seconds

System Integrity Protection: enabled

Notes:
thread_get_state(PAGEIN) returned 0x10000003: (ipc/send) invalid destination port
thread_get_state(EXCEPTION) returned 0x10000003: (ipc/send) invalid destination port
thread_get_state(FLAVOR) returned 0x10000003: (ipc/send) invalid destination port

Crashed Thread:        19  Thread Pool Worker

Exception Type:        EXC_BAD_ACCESS (SIGABRT)
Exception Codes:       UNKNOWN_0xD at 0x0000000000000000
Exception Codes:       0x000000000000000d, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

VM Region Info: 0 is not in any region.  Bytes before following region: 4434386944
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      __TEXT                      1084f6000-1088d5000    [ 3964K] r-x/rwx SM=COW  ...n/mono-sgen64

ID   Vend/Dev
8126 731f1002
Seconds Ago   ID   Type
   690000.0   8126 Attach`

**Comment by Taloth** [comment]
@whodean Looks similar as ekko's crash. But yeah, I need trace logs from sonarr. 

It can be either the mediainfo library or sqlite library. Maybe the sonarr logs will tell us what it did just prior to the crash.

After grabbing the logs, next step would be to delete/rename libmediainfo.0.dylib temporarily, disabling sonarr's ability to check mediainfo.

I'm suddenly really happy I went for the cautionary approach and not push straight to develop :smile:

PS: I won't be able to respond the next 21h, sleep and work.

**Comment by whodean** [comment]
Not sure if I'm looking at the right log, but I see this from shortly before the crashing a file called Sonarr.txt in the logs folder:

`
2021-11-06 22:54:41.9|Warn|AggregationService|Object reference not set to an instance of an object

[v3.0.6.1374] System.NullReferenceException: Object reference not set to an instance of an object
  at NzbDrone.Core.Parser.ParsingService.GetEpisodes (NzbDrone.Core.Parser.Model.ParsedEpisodeInfo parsedEpisodeInfo, NzbDrone.Core.Tv.Series series, System.Boolean sceneSource, NzbDrone.Core.IndexerSearch.Definitions.SearchCriteriaBase searchCriteria) [0x00017] in M:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\Parser\ParsingService.cs:203 
  at NzbDrone.Core.MediaFiles.EpisodeImport.Aggregation.Aggregators.AggregateEpisodes.GetEpisodes (NzbDrone.Core.Parser.Model.LocalEpisode localEpisode) [0x00069] in M:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\MediaFiles\EpisodeImport\Aggregation\Aggregators\AggregateEpisodes.cs:83 
  at NzbDrone.Core.MediaFiles.EpisodeImport.Aggregation.Aggregators.AggregateEpisodes.Aggregate (NzbDrone.Core.Parser.Model.LocalEpisode localEpisode, NzbDrone.Core.Download.DownloadClientItem downloadClientItem) [0x00000] in M:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\MediaFiles\EpisodeImport\Aggregation\Aggregators\AggregateEpisodes.cs:22 
  at NzbDrone.Core.MediaFiles.EpisodeImport.Aggregation.AggregationService.Augment (NzbDrone.Core.Parser.Model.LocalEpisode localEpisode, NzbDrone.Core.Download.DownloadClientItem downloadClientItem) [0x000bd] in M:\BuildAgent\work\63739567f01dbcc2\src\NzbDrone.Core\MediaFiles\EpisodeImport\Aggregation\AggregationService.cs:68 

`

**Comment by bakerboy448** [comment]
> Not sure if I'm looking at the right log, but I see this from shortly before the crashing a file called Sonarr.txt in the logs folder:

you're not - those are info; trace is what's needed
https://wiki.servarr.com/sonarr/troubleshooting#logging-and-log-files

**Comment by whodean** [comment]
> > Not sure if I'm looking at the right log, but I see this from shortly before the crashing a file called Sonarr.txt in the logs folder:
> 
> you're not - those are info; trace is what's needed https://wiki.servarr.com/sonarr/troubleshooting#logging-and-log-files

Thank You

**Comment by Taloth** [comment]
You may have to change log level via the config.xml file, given that it crashes quickly.

**Comment by whodean** [comment]
Well, its been up for 3 hours now for me, but I'll look at grabbing that next time if/when it crashes

**Comment by ekko20six** [comment]
Ok I think I have followed all the steps re setting to debug, getting rid of old log files, restart Sonarr and then watching it crash.
Hopefully the two log files help, the [sonarr.txt](https://github.com/Sonarr/Sonarr/files/7495303/sonarr.txt) one is very short at 60 lines and the [sonarr.debug.txt](https://github.com/Sonarr/Sonarr/files/7495301/sonarr.debug.txt) one is 902 - so not massive either.

Holler out what else I can do to help

**Comment by jchhenderson** [comment]
I've been running mac-1 for 36 hours now with no issues at all.
- Sonarr 3.0.6.1374
- mono 6.4.0.198
- macOS 12.0.1 (Mac mini 2018)

No clue why it is working for me and not for others, but my mono release is different to ekko20six

**Comment by bakerboy448** [comment]
> Ok I think I have followed all the steps re setting to debug, getting rid of old log files, restart Sonarr and then watching it crash. Hopefully the two log files help, the [sonarr.txt](https://github.com/Sonarr/Sonarr/files/7495303/sonarr.txt) one is very short at 60 lines and the [sonarr.debug.txt](https://github.com/Sonarr/Sonarr/files/7495301/sonarr.debug.txt) one is 902 - so not massive either.
> 
> Holler out what else I can do to help
> 
> EDIT - whoops didn't do to trace - will update in a moment EDIT 2 - ok changed to Trace in the xml file and now it hasn't crashed... will continue to monitor and report back EDIT 3 - ok so we crashed again at 7:23 [sonarr.txt](https://github.com/Sonarr/Sonarr/files/7495405/sonarr.txt), [sonarr.debug.txt](https://github.com/Sonarr/Sonarr/files/7495411/sonarr.debug.txt), [sonarr.trace.txt](https://github.com/Sonarr/Sonarr/files/7495414/sonarr.trace.txt). The debug got up to 3 and trace got up to 8 but I assume you just want the one closest to the crash?

Those trace logs don't show any crash.

please see the wiki for how to provide log files.  Generally in this case - we only need trace and several hundred lines around either end of the crash.   You are also doing explicitly what the wiki says not to 🤷‍♂️ https://wiki.servarr.com/sonarr/troubleshooting#logging-and-log-files

**Comment by jchhenderson** [comment]
cool.  This might also explain the difference between ekko20six crashing and my not.

@ekko20six can you see which version of libsqlite you are running?  I assume is isn't libsqlite3.dylib and instead is whatever version is loading on your Mac from some other app's install.

**Comment by ekko20six** [comment]
> 
> Those trace logs don't show any crash? (Disclaimer on mobile).
> 
> please see the wiki for how to provide log files. Generally in this case - we only need trace and several hundred lines around either end of the crash. You are also doing explicitly what the wiki says not to 🤷‍♂️ https://wiki.servarr.com/sonarr/troubleshooting#logging-and-log-files

@bakerboy448 you know I am just trying to help out here and grabbing what I think I am being asked to grab to help figure out the issue, far easier for me to just do nothing and wait for a new version....
I did notice in the files that I wasn't seeing anything jump out at me which is why I gave the time of the crash and the files I uploaded with the question of is this is what's wanted. More flies with honey as the saying goes and in this case your sass was the opposite

**Comment by Taloth** [comment]
100%, Which is why I replied to it immediately to correct it.

Anyway, new version is out. Should be in System->Updates. After install please check if both libsqlite and libmediainfo exist.

Thanks for all your support so far, it really helps to iron out bumps.

**Comment by ekko20six** [comment]
> 100%, Which is why I replied to it (at work) immediately to correct it.
> 
> Anyway, new version is out. Should be in System->Updates. After install please check if both libsqlite and libmediainfo exist.
> 
> Thanks for all your support so far, it really helps to iron out bumps.

Thanks @Taloth  - I am still reading the following replies and following steps such as grabbing console, checking libmediainfo etc.. does it help to still do that before updating or just update and see how it goes?

**Comment by ekko20six** [comment]
> cool. This might also explain the difference between ekko20six crashing and my not.
> 
> @ekko20six can you see which version of libsqlite you are running? I assume is isn't libsqlite3.dylib and instead is whatever version is loading on your Mac from some other app's install.

Sorry @jchhenderson - can you give me some steps to follow to get that output to confirm which libsqlite3.dylib version is coming up? I tried and failed to follow your post :) 
I did check the bin folder and like you there is no libsqlite3.dylib though so I assume I will get the whatever other version my system has?

**Comment by Taloth** [comment]
@ekko20six not needed, go straight for the update. The exact system version is irrelevant, it's super old.

**Comment by ekko20six** [comment]
> @ekko20six not needed, go straight for the update. The exact system version is irrelevant, it's super old.

phew! I was googling the 'DYLD_PRINT_LIBRARIES environment variable' and my morning coffee has not kicked in enough to push through the brain fog.
Have updated and its updating all the shows. I'll get ready for work (have to head into the office and not work from home) so will check for a crash just before I leave or report back after work.

Also, @Taloth , thank you for your understanding and direct replies, I do appreciate your time and encouragement on what steps to follow to help out.

**Comment by jchhenderson** [comment]
> Sorry @jchhenderson - can you give me some steps to follow

no need to apologise :-)

**Comment by jchhenderson** [comment]
OK, updated to 3.0.6.1376 and can see libsqlite now...
![Screenshot 2021-11-08 at 23 35 05](https://user-images.githubusercontent.com/34644672/140835051-97c3dabb-dd23-45a5-8c79-fa983595b8b1.png)

**Comment by ekko20six** [comment]
> DYLD_PRINT_LIBRARIES=1 mono

ahhh!! I was adding in the Sonarr part after that!! that's why it didn't work, thanks :) good to know in the future perhaps 

@Taloth just back from the office and Sonarr is still happily humming away nicely with no crash all day.

**Comment by jchhenderson** [comment]
@ekko20six 👍 

@Taloth, yep, Sonarr humming away for me too!  I will try an import shortly and will hopefully confirm that as well.

**Comment by Taloth** [comment]
Good news I guess. We should just let it hum for a day or two. To be sure. Thx so far for the assist.

**Comment by whodean** [comment]
So far so good, great work @Taloth 

```
About
Version
3.0.6.1376
Mono Version
6.12.0.122
AppData directory
/Users/mini/.config/Sonarr
Startup directory
/Applications/Sonarr.app/Contents/MacOS/bin
Mode
Console
Uptime
1d17hours
```

**Comment by ekko20six** [comment]
All good my end still too, still humming away nicely. 
Not sure which import you all are referring to but I’ve recently started manual import/move for post processed files and that’s working well. All automated tasks working without crashing too. 
top job 🙂

**Comment by Taloth** [comment]
I'll push it to develop branch either today or tomorrow. Thx again for testing.

**Comment by Taloth** [comment]
hm, I just realized, both of you are on intel. we haven't actually tested m1 yet.

**Comment by ekko20six** [comment]
I am indeed, and unfortunately I don't have any m1 Mac yet... waiting for the 27" iMac version (and the new MacBook Pro dropped the Touch Bar...) so apologies that I cant test on an m1 machine

**Comment by jchhenderson** [comment]
@Taloth I have access to an M1 - let me test it tomorrow for you.

**Comment by jchhenderson** [comment]
had a quick go.  Sonarr is running but I cannot open the page at http://localhost:8989

Will try again tomorrow.

**Comment by Taloth** [comment]
Awesome, please lemme know w hat the output is when you run sonarr from the cmdline with `DYLD_PRINT_LIBRARIES=1 Sonarr`. That should log dylib loading and maybe report any crashes.

**Comment by jchhenderson** [comment]
`DYLD_PRINT_LIBRARIES=1 Sonarr` gives me `command not found: Sonarr`
mono v6.12.0.122
MacOS 12.0.1

I don't have it on the `mac-m1` release channel yet so I may be in a vicious circle?

**Comment by Taloth** [comment]
You can install it manually from here `https://download.sonarr.tv/v3/mac-m1/3.0.6.1376/`.

Also the command must be run in a terminal while in the directory where the sonarr binaries are. Somewhere in  your applications folder.

**Comment by jchhenderson** [comment]
OK. it's running and showing usage of CPU time (in Activity monitor) however Safari still cannot connect to http://localhost:8989.

`DYLD_PRINT_LIBRARIES=1 Sonarr` doesn't work either on my m1 Mac or Intel Mac (which is running Sonarr just fine).

**Comment by Taloth** [comment]
You should check the regular sonarr logs or console.

You already used DYLD_* earlier, it's the same thing. You should be able to make it work :smile:

**Comment by jchhenderson** [comment]
> You already used DYLD_* earlier, it's the same thing. You should be able to make it work 😄

to be explicit:
- `DYLD_PRINT_LIBRARIES=1 Mono` works on both m1 and Intel
- `DYLD_PRINT_LIBRARIES=1 Sonarr` returns the error `command not found: Sonarr ` on both machines.

**Comment by jchhenderson** [comment]
> You should check the regular sonarr logs or console.

I cannot access ~/.config/... remotely so will check when I am on site next weekend.
Console shows nothing of use regarding Sonarr.

**Comment by ewancluckie** [comment]
For what it's worth, I'm running the latest production version of Sonarr on M1 Mac Mini with Monterey and it's running great. I've only experienced one crash since the 1st of November.

**Comment by jchhenderson** [comment]
> For what it's worth, I'm running the latest production version of Sonarr on M1 Mac Mini with Monterey and it's running great.

Actually, that's useful - thanks!
I will download the production release and see if it works for me as well 👍

**Comment by jchhenderson** [comment]
OK, still doesn't work for me so I have got something else going on here...

**Comment by Taloth** [comment]
> > You already used DYLD_* earlier, it's the same thing. You should be able to make it work 😄
> 
> to be explicit:
> 
> * `DYLD_PRINT_LIBRARIES=1 Mono` works on both m1 and Intel
> * `DYLD_PRINT_LIBRARIES=1 Sonarr` returns the error `command not found: Sonarr ` on both machines.

@jchhenderson 

 `DYLD_PRINT_LIBRARIES=1 ./Sonarr` in the Sonarr app directory (`/Applications/Sonarr.app/Contents/MacOS/`). Note it's not about whether it works, it's about which libraries get loaded.

**Comment by wampa1** [comment]
The silent shutdown continues to be a problem for me.

Here's some relevant stuff from the Console log:

-------------------------------------
Translated Report (Full Report Below)
-------------------------------------
Process: mono-sgen64 [53026]
Path: /Applications/Sonarr.app/Contents/MacOS/Sonarr
Identifier: mono-sgen64
Version: ???
Code Type: X86-64 (Translated)
Parent Process: launchd [1]
User ID: 501
Date/Time: 2021-11-20 05:45:50.7991 -0500
OS Version: macOS 12.0.1 (21A559)
Report Version: 12
Anonymous UUID: DBCA941D-EBA0-0C9B-B016-066A3DA21F74
Time Awake Since Boot: 2100000 seconds
System Integrity Protection: enabled
Notes:
thread_get_state(PAGEIN) returned 0x10000003: (ipc/send) invalid destination port
thread_get_state(EXCEPTION) returned 0x10000003: (ipc/send) invalid destination port
thread_get_state(FLAVOR) returned 0x10000003: (ipc/send) invalid destination port
Crashed Thread: 18 Thread Pool Worker
Exception Type: EXC_BAD_ACCESS (SIGABRT)
Exception Codes: KERN_INVALID_ADDRESS at 0x000000010000003e
Exception Codes: 0x0000000000000001, 0x000000010000003e
Exception Note: EXC_CORPSE_NOTIFY

**Comment by ewancluckie** [comment]
I may have spoken too soon. I've now had 3 more crashes in the last couple of weeks. Always seems to happen in the middle of the night though so I'm still diagnosing whether it is just automatic software updates forcing a reboot of the Mac and then Sonarr not starting up again on boot. Will try to pull some logs the next time it happens. Not confirmed yet if it is a Sonarr issue at my end, but Radarr seems to be having no issues.

**Comment by phattrance** [comment]
Im on Mac M1 and the sonarr keep crashing a few times per day. Really annoying.

**Comment by bakerboy448** [comment]
We're aware it's an issue.

please keep comments on topic and relevant to testing or fixing it.

**Comment by Taloth** [comment]
ewan, lemme know if it happens again with some logs. Probably gonna be a while so I might just push this to develop since it's an improvement regardless.

**Comment by wampa1** [comment]
I've moved to the mac-m1 branch and have had no issues since, up for > 7 days now.

Huge appreciation and brilliant work.

**Comment by Loneshade** [comment]
deleting and replacing the the sqlite 3 file with every update is annoying and still does not fix the issue. As my Sonarr is still dropping out. 

	Native stacktrace:
=================================================================
	0x100e3d5c9 - /Library/Frameworks/Mono.framework/Versions/6.12.0/bin/mono-sgen64 : mono_dump_native_crash_info
	0x100dd537e - /Library/Frameworks/Mono.framework/Versions/6.12.0/bin/mono-sgen64 : mono_handle_native_crash
	0x100e37746 - /Library/Frameworks/Mono.framework/Versions/6.12.0/bin/mono-sgen64 : altstack_handle_and_restore
	0x7ff80fdd6618 - /usr/lib/libsqlite3.dylib : sqlite3DbMallocRawNN
	0x7ff80fe03191 - /usr/lib/libsqlite3.dylib : resizeResolveLabel
	0x7ff80fd7b5e8 - /usr/lib/libsqlite3.dylib : sqlite3GenerateConstraintChecks
	0x7ff80fd75f96 - /usr/lib/libsqlite3.dylib : sqlite3Insert
	0x7ff80fd15094 - /usr/lib/libsqlite3.dylib : yy_reduce
	0x7ff80fd11695 - /usr/lib/libsqlite3.dylib : sqlite3RunParser
	0x7ff80fd10b31 - /usr/lib/libsqlite3.dylib : sqlite3Prepare
	0x7ff80fd10678 - /usr/lib/libsqlite3.dylib : sqlite3LockAndPrepare
	0x7ff80fd85fa2 - /usr/lib/libsqlite3.dylib : sqlite3_prepare_v2
	0x105edb4ff - Unknown
	Managed Stacktrace:
=================================================================
	  at <unknown> <0xffffffff>
	  at System.Data.SQLite.UnsafeNativeMethods:sqlite3_prepare_v2 <0x000be>
	  at System.Data.SQLite.SQLite3:Prepare <0x00612>
	  at System.Data.SQLite.SQLiteCommand:BuildNextCommand <0x002b6>
	  at System.Data.SQLite.SQLiteCommand:GetStatement <0x0005a>
	  at System.Data.SQLite.SQLiteCommand:GetStatement <0x000aa>
	  at System.Data.SQLite.SQLiteDataReader:NextResult <0x0072a>
	  at System.Data.SQLite.SQLiteDataReader:.ctor <0x00344>
	  at System.Data.SQLite.SQLiteDataReader:.ctor <0x00082>
	  at System.Data.SQLite.SQLiteCommand:ExecuteReader <0x0008a>
	  at System.Data.SQLite.SQLiteCommand:ExecuteNonQuery <0x0006a>
	  at System.Data.SQLite.SQLiteCommand:ExecuteNonQuery <0x00052>
	  at NzbDrone.Core.Instrumentation.DatabaseTarget:Write <0x00847>
	  at NLog.Targets.Target:Write <0x00052>
	  at NLog.Targets.Target:Write <0x0009b>
	  at NLog.Targets.Target:Write <0x00043>
	  at NLog.Targets.Target:WriteAsyncThreadSafe <0x0021c>
	  at NLog.Targets.Target:WriteAsyncLogEvents <0x007d6>
	  at NLog.Targets.Target:WriteAsyncLogEvents <0x0005a>
	  at NLog.Targets.Wrappers.AsyncTargetWrapper:WriteEventsInQueue <0x0014a>
	  at NLog.Targets.Wrappers.AsyncTargetWrapper:ProcessPendingEvents <0x00102>
	  at Scheduler:TimerCB <0x00142>
	  at System.Threading.QueueUserWorkItemCallback:System.Threading.IThreadPoolWorkItem.ExecuteWorkItem <0x00042>
	  at System.Threading.ThreadPoolWorkQueue:Dispatch <0x00279>
	  at System.Threading._ThreadPoolWaitCallback:PerformWaitCallback <0x0001c>
	  at <Module>:runtime_invoke_bool <0x000a5>
=================================================================
zsh: abort      mono /Applications/Sonarr.app/Contents/MacOS/Sonarr.exe

**Comment by bakerboy448** [comment]
@Loneshade theres an entire branch to check and try instead of the file.

Did you not see the numerous comments on that?

**Comment by Loneshade** [comment]
@bakerboy448 Yeah I have been following it and I am on the developer branch, unless I have missed a comment on this thread somewhere. I have been on the developer branch since V3 was a beta and I haven't changed it since. I have noticed that my Sonarr application didn't have a bin folder so I have completely deleted the the app and reinstalled it. Replacing the Sqlite 3 works sometimes but isn't a fix, and I have been trying the above with the media libraries as well.

**Comment by bakerboy448** [comment]
@Loneshade https://github.com/Sonarr/Sonarr/issues/4718#issuecomment-962496092

tho I think @Taloth was just going to merge this to develop since it'll fix it for many users.

no idea when that will actually occur

**Comment by df-a** [comment]
> This should fix it

This fixed it for me

Intel Mini 6,2 running Monterrey

**Comment by rjhollan** [comment]
I've been running Sonar 3.0.6.1342 with Mono 6.12.0.122 for the past ~3 weeks on an M1 MBP (16", M1 MAX, 64GB) and it randomly crashes silently.  I just updated to 3.0.6.1404 so we'll see if that's more stable and report back....

**Comment by Taloth** [comment]
Use the regular `develop` branch (atm 3.0.6.1432) instead, I merged it a while back. The app should auto-update to develop btw, if you're on the mac-m1 branch and have auto-update enabled. Which is why I didn't mention it earlier.

**Comment by rjhollan** [comment]
Thanks!  I switched from main with no auto-update to develop with updates, re-ran the update check task, and it's now running 3.0.6.1432 from the develop branch.  I'm on OSX 12.0.1 for now, probably go to 12.1 soon.

**Comment by StankDawg** [comment]
Still an issue for me.  I updated the file above, renamed the old one, and it did seem to run fine for a week or so.  I DID NOT upgrade it, and I thought I also saw mention that this solution was rolled into the main branch anyway by now?

This is clearly a MONO issue.  When I check my OSX logs, the only crash that I see is a MONO crash every time I notice that sonarr disconnects.

I can share these crash files if anyone wants to see them.  I wish I was better at reading them myself.

FWIW:  

Version
3.0.6.1432
Mono Version
6.12.0.122
